ninja-core/src/test/java/ninja/AuthenticityFilterTest.java

```diff
@@ -0,0 +1,73 @@
+package ninja;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+import ninja.session.Session;
+import ninja.utils.NinjaConstant;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AuthenticityFilterTest {
+    
+    @Mock
+    private Context context;
+    
+    @Mock
+    private FilterChain filterChain;
+    
+    @Mock
+    private Session session;
+    
+    @Mock
+    private NinjaDefault ninjaDefault;
+    
+    Result forbidden = Results
+            .forbidden()
+            .supportedContentTypes(Result.TEXT_HTML, Result.APPLICATION_JSON, Result.APPLICATION_XML)
+            .fallbackContentType(Result.TEXT_HTML)
+            .render("")
+            .template("");
+    
+    Result ok = Results
+            .ok()
+            .supportedContentTypes(Result.TEXT_HTML, Result.APPLICATION_JSON, Result.APPLICATION_XML)
+            .fallbackContentType(Result.TEXT_HTML)
+            .render("")
+            .template("");
+    
+    private AuthenticityFilter authenticityFilter;
+
+    @Before
+    public void init() {
+        authenticityFilter = new AuthenticityFilter(ninjaDefault);        
+    }
+    
+    @Test
+    public void testAuthenticityFail() {
+        when(context.getParameter(NinjaConstant.AUTHENTICITY_TOKEN)).thenReturn("foo");
+        when(context.getSession()).thenReturn(session);
+        when(context.getSession().getAuthenticityToken()).thenReturn("bar");
+        when(ninjaDefault.getForbiddenResult(context)).thenReturn(forbidden);
```

Comment:
you can simplify the code a bit... we don't need ok and forbidden results...

Just do a check  Mockito.verify(ninjaDefault).getForbiddenResult(context)

or

Mockito.verify(filterChain).next(context);

assertNotNull and assertEquals is not needed imho...

That makes the code a bit shorter...